clients/ts/signalr/src/LongPollingTransport.ts

```diff
@@ -177,29 +164,37 @@ export class LongPollingTransport implements ITransport {
     }
 
     public async stop(): Promise<void> {
-        // Send a DELETE request to stop the poll
+        this.logger.log(LogLevel.Trace, `(LongPolling transport) Stopping polling.`);
+
+        // Tell receiving loop to stop, abort any current request, and then wait for it to finish
+        this.running = false;
+        this.pollAbort.abort();
+
         try {
-            this.running = false;
+            await this.receiving;
```

Comment:
I recall having problem making this work well because transports were reusable. You should probably copy all of the state to locals before doing anything.

Reply:
What happens if you don't await stop

Reply:
A transport isn't reused when a HubConnection is started and stopped. And the HttpConnection state is immediately set to Disconnected when stop is called.

Reply:
Ah! I forgot @anurse did that work